src/components/FeaturePanel/PublicTransport/requestRoutes.ts

```diff
@@ -1,43 +1,41 @@
-import { fetchText } from '../../../services/fetch';
+import { fetchJson } from '../../../services/fetch';
+import { overpassGeomToGeojson } from '../../../services/overpassSearch';
 
 export interface LineInformation {
   ref: string;
   colour: string | undefined;
+  service: string | undefined;
 }
 
-export async function requestLines(
-  featureType: 'node' | 'way' | 'relation',
-  id: number,
-) {
-  // use the overpass api to request the lines in
-  const overpassQuery = `[out:csv(ref, colour; false; ';')];
+export async function requestLines(featureType: string, id: number) {
+  const overpassQueryJson = `[out:json];
   ${featureType}(${id});
   rel(bn)["public_transport"="stop_area"];
-node(r: "stop") -> .stops;
+  node(r: "stop") -> .stops;
     rel(bn.stops)["route"~"bus|train|tram|subway|light_rail|ferry|monorail"];
-    out;`;
+    out geom qt;`;
 
-  // send the request
-  const response: string = await fetchText(
+  const geoJson = await fetchJson(
     `https://overpass-api.de/api/interpreter?data=${encodeURIComponent(
-      overpassQuery,
+      overpassQueryJson,
     )}`,
-  );
+  ).then(overpassGeomToGeojson);
 
-  const resData = response
-    .split('\n')
-    .map((line) => {
-      const ref = line.split(';').slice(0, -1).join(';');
-      let colour = line.split(';')[line.split(';').length - 1];
-
-      // set colour to undefined if it is empty
-      if (colour === '') colour = undefined;
-      return { ref, colour } as LineInformation;
-    })
+  const resData = geoJson
+    .map(
+      ({ properties }): LineInformation => ({
+        ref: `${properties.ref || properties.name}`,
+        colour: properties.colour?.toString() || undefined,
+        service: properties.service?.toString() || undefined,
+      }),
+    )
     .sort((a, b) => a.ref.localeCompare(b.ref, undefined, { numeric: true }))
     .filter(({ ref }) => ref !== '');
 
-  return resData.filter(
-    (line, index) => resData.findIndex((l) => l.ref === line.ref) === index,
-  );
+  return {
+    geoJson: { type: 'FeatureCollection', features: geoJson },
+    routes: resData.filter(
+      (line, index) => resData.findIndex((l) => l.ref === line.ref) === index,
```

Comment:
What does this filter do? Can you please extract it a variable and call it apropriatly (if it is more difficult, then please add comment)

Reply:
This filter was already there in the original version. It removes duplicates.

Reply:
I now replaced it with lodash's `uniqBy` function